client/src/main/scala/ch/epfl/bluebrain/nexus/iam/client/IamClient.scala

```diff
@@ -31,46 +31,51 @@ class IamClient[F[_]] private[client] (config: IamClientConfig,
 
   private val log = Logger[this.type]
 
-  /**
-    * Retrieve the ''caller'' from the implicitly optional [[AuthToken]]
-    *
-    */
-  def getCaller(implicit credentials: Option[AuthToken]): F[Caller] =
-    credentials
-      .map { _ =>
-        callerClient(requestFrom(config.prefix / "oauth2" / "user"))
-          .recoverWith { case e => recover(e, config.prefix / "oauth2" / "user") }
-      }
-      .getOrElse(F.pure(Caller.anonymous))
+  object acls {
 
-  /**
-    * Retrieve the current ''acls'' for some particular ''path''.
-    *
-    * @param path        the target resource
-    * @param ancestors   matches only the exact ''path'' (false) or its ancestors also (true)
-    * @param self        matches only the caller identities
-    * @param credentials an optionally available token
-    */
-  def getAcls(path: Path, ancestors: Boolean = false, self: Boolean = false)(
-      implicit credentials: Option[AuthToken]): F[AccessControlLists] = {
-    val req =
-      requestFrom(path :: (config.prefix / "acls"), Query("ancestors" -> ancestors.toString, "self" -> self.toString))
-    aclsClient(req).recoverWith { case e => recover(e, path) }
+    /**
+      * Retrieve the current ''acls'' for some particular ''path''.
+      *
+      * @param path        the target resource
+      * @param ancestors   matches only the exact ''path'' (false) or its ancestors also (true)
+      * @param self        matches only the caller identities
+      * @param credentials an optionally available token
+      */
+    def list(path: Path, ancestors: Boolean = false, self: Boolean = false)(
+        implicit credentials: Option[AuthToken]): F[AccessControlLists] = {
+      val req =
+        requestFrom(path :: (config.prefix / "acls"), Query("ancestors" -> ancestors.toString, "self" -> self.toString))
+      aclsClient(req).recoverWith { case e => recover(e, path) }
+    }
+
+    /**
+      * Checks the presence of a specific ''permission'' on a particular ''path''.
+      *
+      * @param path        the target resource
+      * @param permission  the permission to check
+      * @param credentials an optionally available token
+      */
+    def authorizeOn(path: Path, permission: Permission)(implicit credentials: Option[AuthToken]): F[Unit] =
```

Comment:
I'm not sure that this method is part of `acls`, I'd leave it directly on the client.